src/pages/trader-tools/_margin-calculator.js

```diff
@@ -0,0 +1,442 @@
+        const volume_error = validation.volume(values.volume)
+        const assetPrice_error = validation.assetPrice(values.assetPrice)
+        const leverage_error = validation.leverage(values.leverage)
+
+        if (symbol_error) {
+            errors.symbol = symbol_error
+        }
+        if (volume_error) {
+            errors.volume = volume_error
+        }
+        if (assetPrice_error) {
+            errors.assetPrice = assetPrice_error
+        }
+        if (leverage_error) {
+            errors.leverage = leverage_error
+        }
+
+        return errors
+    }
+
+    const getMarginCurrency = (symbol) => {
+        let currency = 'USD'
+        if (symbol.market === 'synthetic_indices') {
+            currency = 'USD'
+        } else {
+            if (symbol.market === 'commodities') {
+                currency = 'USD'
+            } else {
+                if (symbol.name === 'DAX_30') {
+                    currency = 'EUR'
+                } else {
+                    if (symbol.displayName !== 'Please select a value')
+                        currency = symbol.displayName.slice(-3)
+                }
+            }
+        }
+        return currency
+    }
+
+    const getContractSize = (symbol) => {
+        let contractSize = 1
+
+        if (symbol.market === 'forex') {
+            contractSize = 1000
+        } else if (symbol.displayName === 'Silver/USD') {
+            contractSize = 500
+        } else if (
+            symbol.displayName === 'Gold/USD' ||
+            symbol.displayName === 'Palladium/USD' ||
+            symbol.displayName === 'Platinum/USD'
+        ) {
+            contractSize = 100
+        } else if (symbol.displayName === 'Step Index') {
+            contractSize = 10
+        }
+
+        return contractSize
+    }
+
+    useEffect(() => {
+        // fetchSymbol()
+        // console.log('effect')
+    }, [])
+
+    return (
+        <Container direction="column">
+            <StyledHeaderTitle align="center" as="h2">
+                {localize('Margin calculator')}
+            </StyledHeaderTitle>
+            <StyledHeaderP>
+                <Localize
+                    translate_text="Our margin calculator helps you to estimate the margin required to keep your
+        positions open overnight on Deriv MetaTrader 5 (DMT5)."
+                    components={[<strong key={0} />]}
+                />
+            </StyledHeaderP>
+            <WrapContainer mb="4.0rem">
+                <StyledFormWrapper>
+                    <Formik
+                        initialValues={{
+                            accountType: 'Synthetic',
+                            margin: '',
+                            marginSymbol: 'USD',
+                            symbol: '',
+                            volume: '',
+                            assetPrice: '',
+                            leverage: '',
+                            optionList: syntheticItemLists,
```

Comment:
will option list change on change of something?

Reply:
yes , the dropdown list for symbol